File: core/core-api/src/main/java/com/coffee/gu/api/controller/v1/response/order/OrderCheckoutResponse.java
package com.coffee.gu.api.controller.v1.response.order;

import com.coffee.gu.api.controller.v1.response.IssuedCouponResponse;
import com.coffee.gu.coupon.IssuedCoupon;
import com.coffee.gu.order.Order;

import java.math.BigDecimal;
import java.util.List;

public record OrderCheckoutResponse(
        String key,
        String name,
        BigDecimal totalPrice,
        List<OrderLineResponse> lines,
        List<IssuedCouponResponse> usableCoupons
) {
    public static OrderCheckoutResponse of(Order order, List<IssuedCoupon> issuedCoupons) {
        return new OrderCheckoutResponse(
                order.getKey(),
                order.getName(),
                order.getTotalPrice(),
                order.getLines()
                        .stream()
                        .map(line -> new OrderLineResponse(
                                line.getMenuId(),
                                line.getMenuName(),
                                line.getImageUrl(),
                                line.getDescription(),
                                line.getQuantity(),
                                line.getUnitPrice(),
                                line.getTotalPrice()
                        )).toList(),
                issuedCoupons.stream().map(IssuedCouponResponse::from).toList()
        );
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/api/controller/v1/response/order/OrderCheckoutResponse.kt
package com.coffee.gu.api.controller.v1.response.order

import com.coffee.gu.api.controller.v1.response.IssuedCouponResponse
import com.coffee.gu.coupon.IssuedCoupon
import com.coffee.gu.order.Order
import java.math.BigDecimal

class OrderCheckoutResponse(
    val key: String,
    val name: String,
    val totalPrice: BigDecimal,
    val lines: List<OrderLineResponse>,
    val usableCoupons: List<IssuedCouponResponse>,
) {
    companion object {
        @JvmStatic
        fun of(order: Order, issuedCoupons: List<IssuedCoupon>): OrderCheckoutResponse {
            return OrderCheckoutResponse(
                key = order.key,
                name = order.name,
                totalPrice = order.totalPrice,
                lines = order.lines.map { line ->
                    OrderLineResponse(
                        menuId = line.menuId,
                        menuName = line.menuName,
                        imageUrl = line.imageUrl ?: "",
                        description = line.description ?: "",
                        quantity = line.quantity,
                        unitPrice = line.unitPrice,
                        totalPrice = line.totalPrice
                    )
                },
                usableCoupons = IssuedCouponResponse.from(issuedCoupons)
            )
        }
    }
}
